.konflux/scripts/render_snapshot.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import time
+
+
+def load_image_refs():
+    return json.loads(os.getenv("IMAGE_REFS", '[]'))
+
+
+def process_component(component, name_suffix):
+    if name_suffix != "":
+        name = f"{component['component']}-{name_suffix}"
+    else:
+        name = component["component"]
+    return {
+        "containerImage": component["ref"],
+        "name": name,
+        "source": {
+            "git": {
+                "revision": component["revision"],
+                "url": component["repository"]
+            }
+        }
+    }
+
+
+def construct_snapshot(
+    snapshot_name_prefix,
+    pipeline_run_name,
+    application,
+    components
+):
+    snapshot_name = f"{snapshot_name_prefix}-{int(time.time())}"
+    return {
+        "apiVersion": "appstudio.redhat.com/v1alpha1",
+        "kind": "Snapshot",
+        "metadata": {
+            "name": snapshot_name,
+            "labels": {
+                "appstudio.openshift.io/build-pipelinerun": pipeline_run_name
+            }
+        },
+        "spec": {
+            "application": application,
+            "components": components
+        }
+    }
+
+
+def determine_component_name_suffix(application):
+    return application.lstrip("acs-")
```

Comment:
Is the idea to take the version suffix like `-4-6` from `acs-4-6` and apply it to components? I think, it may need some better description (either in comments or in function name, or both) for the future generations. For the function name, I can suggest `determine_version_suffix` or `get_version_suffix`.

Also, extracting optional suffix may look more self-explanatory than stripping prefix. Yes, that'll likely require a, regular expression but I think the clarity may worth it.

Reply:
Done in https://github.com/stackrox/stackrox/pull/13577/commits/7f3b94edc43cc0fb3fb17fd9d08d86f555d466f7